core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -279,6 +279,9 @@ public void listStatus() throws Exception {
     FileStatus[] fileStatuses = alluxioHadoopFs.listStatus(path);
     assertFileInfoEqualsFileStatus(fileInfo1, fileStatuses[0]);
     assertFileInfoEqualsFileStatus(fileInfo2, fileStatuses[1]);
+    if (alluxioHadoopFs != null) {
+      alluxioHadoopFs.close();
```

Comment:
Sorry, it looks like a secondary bug is getting in the way. It should be simple to fix though.

It looks like the problem is in the `close` method of `AbstractFileSystem`. @jiaosummer, could you add a null check in `AbstractFileSystem.java`? Currently we do
```java
  @Override
  public void close() throws IOException {
    if (mContext != FileSystemContext.INSTANCE) {
      mContext.close();
    }
    super.close();
  }
```
we should change the first line to
```java
    if (mContext != null && mContext != FileSystemContext.INSTANCE) {
```
Then we can safely call `alluxioHadoopFs.close()` without worrying about null